tests/sentry/api/endpoints/test_team_details.py

```diff
@@ -49,28 +49,75 @@ def test_simple(self):
 
 
 class TeamDeleteTest(APITestCase):
+
+    def assert_team_deleted(self, team_id, mock_delete_team, transaction_id):
+        """Checks team status, membership in DeletedTeams table, org
+           audit log, and to see that delete function has been called"""
+
+        team = Team.objects.get(id=team_id)
+
+        assert team.status == TeamStatus.PENDING_DELETION
+
+        deleted_team = DeletedTeam.objects.get(slug=team.slug)
+        # in spite of the name, this checks the DeletedTeam object, not the
+        # audit log
+        self.assert_valid_deleted_log(deleted_team, team)
+
+        # *this* actually checks the audit log
+        audit_log_entry = AuditLogEntry.objects.filter(
+            event=AuditLogEntryEvent.TEAM_REMOVE,
+            target_object=team.id
+        )
+        assert audit_log_entry
```

Comment:
If you don't need audit_log_entry later on, you could do a `.get()` here and assert the query does not `.exist()` in the next method, assert_team_not_deleted.